### PR TITLE
Prime number checker

### LINKED ISSUE #6: Prime Number Checker
| **Field**        | **Details**                                                                                                         |
|------------------|---------------------------------------------------------------------------------------------------------------------|
| **Name**         | New Challenge                                                                                                       |
| **Description**  | Create a program that checks if a number is prime or not.                                                           |
| **Title**        | Prime Number Checker                                                                                               |

### PR DESCRIPTION
**Pull request by Husayn01**
---
name: Prime number checker
about: The function checks if the number is a prime
---

## Behavior

- The function checks if the number is a prime by validating its type and checking divisibility rules.
- It returns True if the number is prime and False if it is not.
- The function handles invalid inputs (non-integer values) by raising a TypeError.
- 
### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations
- [x] The function is not called in the function file

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] There are no commented lines of code
- [x] There are no `print` statements anywhere
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/prime_number_checker.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+A module for checking if a number is prime.
+
+Module contents: check if an input number is a prime number.
```
Comment (NoorelsalamAlmakki) [inline solutions/prime_number_checker.py]: Instead of "Module contents: ", using the function name is better.

File: solutions/prime_number_checker.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        False
+        >>> prime_number_checker(0)
+        False
+        >>> prime_number_checker(-5)
+        False
+    """
+    if not isinstance(number, int):
+        raise TypeError("Input must be an integer.")
+
+    if number <= 1:
```
Comment (NoorelsalamAlmakki) [inline solutions/prime_number_checker.py:47]: Include an assertion message.

File: solutions/tests/test_prime_number_checker.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+    def test_largest_two_digit_prime(self):
+        """Verify the largest two-digit prime number (97)."""
+        self.assertTrue(prime_number_checker(97))
+
+    def test_consecutive_numbers_around_prime(self):
+        """Verify behavior with consecutive numbers around a prime."""
+        self.assertFalse(prime_number_checker(8))
+        self.assertTrue(prime_number_checker(7))
+        self.assertFalse(prime_number_checker(6))
```
Comment (NoorelsalamAlmakki) [inline solutions/tests/test_prime_number_checker.py]: This test captures an important aspect of the function behavior. However, the checklist, provided by Evan, states that each test should contain ONLY ONE assertion. I believe this was set to ensure any unexpected behavior is easly identified.

**Review by NoorelsalamAlmakki — COMMENTED** [review]
The code is well organized and the function is thoroughly tested. But, some changes need to be addressed to fully meet the merging criteria. Over all good job!

File: solutions/prime_number_checker.py (hunk 1 above)
Comment (Husayn01) [reply solutions/prime_number_checker.py]: Done

**Comment by Husayn01** [comment]
@NoorelsalamAlmakki, thank you for your feedback, i have resolved all issues and is awaiting review.

**Review by NoorelsalamAlmakki — APPROVED** [review]
